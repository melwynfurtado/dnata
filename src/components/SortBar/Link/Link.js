import React from 'react'
import PropTypes from 'prop-types'

const Link = ({ clickHandle, sortBy, field, label }) => (
  <a className={`nav-item nav-link ${sortBy.key === field ? 'active' : ''}`} href="#" onClick={clickHandle} field={field} order={`${sortBy.order}`}>
    { label }
    { sortBy.key === field && <span className={`ml-1 align-text-top fa fa-sort-${sortBy.order}`} field={field} order={`${sortBy.order}`}></span> }
  </a>  
)

Link.propTypes = {
  clickHandle: PropTypes.func,
  sortBy: PropTypes.shape({
    key: PropTypes.string,
    order: PropTypes.string,
  }),
  field: PropTypes.string,
  label: PropTypes.string,
}

export default Link
