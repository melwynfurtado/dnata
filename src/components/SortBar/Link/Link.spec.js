import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Link from './Link'

describe('Link Component', () => {
  let handleOnClickMock
  let Component

  beforeEach(() => {
    handleOnClickMock = jest.fn()
    Component = <Link 
      field="Stars" 
      label="Stars" 
      clickHandle={handleOnClickMock} 
      sortBy={{key: 'Stars', order: 'desc'}} 
    />
  })

  it('renders correctly when field same as sort by key', () => {
    const tree = renderer
                  .create(Component)
                  .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when field is not same as sort by key', () => {
    Component = <Link 
      field="Distance" 
      label="Distance" 
      clickHandle={handleOnClickMock} 
      sortBy={{key: 'Stars', order: 'desc'}} 
    />    
    const tree = renderer
                  .create(Component)
                  .toJSON()
    expect(tree).toMatchSnapshot()
  })  

  it('calls clickHandle onclick', () => {
    const wrapper = shallow(Component)
    wrapper.simulate('click')
    expect(handleOnClickMock).toHaveBeenCalled()
  })
})