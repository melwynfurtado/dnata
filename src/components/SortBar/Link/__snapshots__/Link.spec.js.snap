// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Component renders correctly when field is not same as sort by key 1`] = `
<a
  className="nav-item nav-link "
  field="Distance"
  href="#"
  onClick={[MockFunction]}
  order="desc"
>
  Distance
</a>
`;

exports[`Link Component renders correctly when field same as sort by key 1`] = `
<a
  className="nav-item nav-link active"
  field="Stars"
  href="#"
  onClick={[MockFunction]}
  order="desc"
>
  Stars
  <span
    className="ml-1 align-text-top fa fa-sort-desc"
    field="Stars"
    order="desc"
  />
</a>
`;
